src/core_plugins/input_control_vis/public/control/control.js

```diff
@@ -7,17 +7,74 @@ Choose a different field or index documents that contain values for this field.`
 }
 
 export class Control {
-  constructor(controlParams, filterManager) {
+  constructor(controlParams, filterManager, kbnApi, useTimeFilter) {
     this.id = controlParams.id;
     this.options = controlParams.options;
     this.type = controlParams.type;
     this.label = controlParams.label ? controlParams.label : controlParams.fieldName;
+    this.useTimeFilter = useTimeFilter;
     this.filterManager = filterManager;
-    this.enable = true;
+    this.kbnApi = kbnApi;
+
     // restore state from kibana filter context
     this.reset();
+    // disable until initialized
+    this.disable('Control has not been initialized');
+  }
+
+  async fetch() {
+    throw new Error('fetch method not defined');
+  }
+
+  /**
+   *
+   * @param ancestors {array of Controls}
+   */
+  setAncestors(ancestors) {
```

Comment:
The `RangeControl` also extends from `Control`, but it doesn't do anything with ancestors, why are we adding all of these here?

Reply:
I wanted to make the ancestor logic as generic as possible so I put it into the Control object. Maybe Range controls will be nestable in a future release.